src/dex-ui/hooks/useInput.tsx

```diff
@@ -12,3 +12,19 @@ export function useInput<T>(initialValue: T) {
     handleChange,
   };
 }
+
```

Comment:
1 - The `useDates` hook can be placed in a separate `useDates.tsx` file. 
2 - We should name this something more specific such as `useDateRange`
3 - There's no need to use generic typing (`<T>`) if the types are already determined. This hook is specifically used for dates so we can explicitly type the parameters as `Date | null`. The only exception is if we plan to use Date like objects from other date libraries.

Reply:
Done